sound/soc/sof/intel/hda-stream.c

```diff
@@ -1246,3 +1246,119 @@ u64 hda_dsp_get_stream_ldp(struct snd_sof_dev *sdev,
 	return ((u64)ldp_u << 32) | ldp_l;
 }
 EXPORT_SYMBOL_NS(hda_dsp_get_stream_ldp, "SND_SOC_SOF_INTEL_HDA_COMMON");
+
+struct hdac_ext_stream *
+hda_dma_prepare(struct device *dev, unsigned int format, unsigned int size,
+		struct snd_dma_buffer *dmab, bool persistent_buffer, int direction,
+		bool is_iccmax, bool pair)
```

Comment:
The generic naming of hda_dma_prepare() / hda_dma_cleanup() I think is confusing.
This is implementing a special use of HD-DMA for data transfer, it does buffer allocation also.

It handles the CL (Code Loader - pair == false)  and a data transfer channel to external device.

Can we find a better name? (sorry)